sklearn/metrics/_plot/roc_curve.py

```diff
@@ -170,7 +170,10 @@ def plot(
         else:
             self.chance_level_ = None
 
-        if "label" in line_kwargs or "label" in chance_level_line_kw:
+        if (
+            line_kwargs.get("label") is not None
+            and chance_level_line_kw.get("label") is not None
```

Comment:
Isn't it an `or` and instead of an `and`? We need one of the two not being None to show the legend?

It also means that we don't have a test right now.

Reply:
Yes, I did change the logic so that the legend is not set as soon as either of the labels is `None`.

The reasoning is that I don't think anyone would set `plot_chance_lvl=True` and override `chance_level_kw` with `{"label": None}`,  so that part could be a bit more flexible. WDYT?

Reply:
For me this is counter-intuitive to have no legend for the chance level curve with the following code:

```python
In [3]: import matplotlib.pyplot as plt
   ...: from sklearn.datasets import make_classification
   ...: from sklearn.metrics import RocCurveDisplay
   ...: from sklearn.model_selection import train_test_split
   ...: from sklearn.svm import SVC
   ...: X, y = make_classification(random_state=0)
   ...: X_train, X_test, y_train, y_test = train_test_split(
   ...:     X, y, random_state=0)
   ...: clf = SVC(random_state=0).fit(X_train, y_train)
   ...: RocCurveDisplay.from_estimator(
   ...:    clf, X_test, y_test, label=None, plot_chance_level=True
   ...: )
   ...: plt.show()
```

If I really want to turn off the legend for the chance level then I can do it with the right params:

```python
In [2]: import matplotlib.pyplot as plt
   ...: from sklearn.datasets import make_classification
   ...: from sklearn.metrics import RocCurveDisplay
   ...: from sklearn.model_selection import train_test_split
   ...: from sklearn.svm import SVC
   ...: X, y = make_classification(random_state=0)
   ...: X_train, X_test, y_train, y_test = train_test_split(
   ...:     X, y, random_state=0)
   ...: clf = SVC(random_state=0).fit(X_train, y_train)
   ...: RocCurveDisplay.from_estimator(
   ...:    clf, X_test, y_test, label=None, plot_chance_level=True, chance_level_kw={"label": None},
   ...: )
   ...: plt.show()
```

So to have this last behaviour you need to have a `or` operator and not a `and` operator.